Terminal.Gui/Views/DateField.cs

```diff
@@ -374,25 +373,25 @@ public override bool MouseEvent (MouseEvent ev)
 		/// Virtual method that will invoke the <see cref="DateChanged"/>  with a <see cref="DateTimeEventArgs"/>.
 		/// </summary>
 		/// <param name="args">The arguments of the <see cref="DateTimeEventArgs"/></param>
-		public virtual void OnDateChanged (DateTimeEventArgs args)
+		public virtual void OnDateChanged (DateTimeEventArgs<DateTime> args)
 		{
-			DateChanged?.Invoke (this, args);
+			DateChanged?.Invoke (args);
 		}
 	}
 
 	/// <summary>
 	/// Handled the <see cref="EventArgs"/> for <see cref="DateField"/> or <see cref="TimeField"/> events.
 	/// </summary>
-	public class DateTimeEventArgs : EventArgs {
+	public class DateTimeEventArgs<T> : EventArgs {
```

Comment:
Can you explain why you used `DateTimeEventArgs<T>` here instead of `DateTimeEventArgs<TimeSpan>` here?

Reply:
Because this `DateTimeEventArgs `is being used simultaneously by `DateField `and `TimeField `and are of different `types`.